### PR TITLE
Support -9e as an argument for xz compression

### PR DESCRIPTION
**Pull request by dtcooper**
Hi there,

This allows for a little more extreme compression with `xz` (which is what I believe Raspberry Pi uses for their distributables).

Thanks!

### CODE REVIEW

File: src/pi-gen-config.ts [hunk 1]
```diff
@@ -92,16 +92,17 @@ export async function validateConfig(config: PiGenConfig): Promise<void> {
...
-  ) {
+  const deployCompression = config.deployCompression?.toLowerCase()
+  if (!['none', 'zip', 'gz', 'xz'].includes(deployCompression)) {
     throw new Error('compression must be one of ["none", "zip", "gz", "xz"]')
   }
 
-  if (!/^[0-9]$/.test(config.compressionLevel)) {
-    throw new Error('compression-level must be between 0 and 9')
+  if (
+    !/^[0-9]$/.test(config.compressionLevel) &&
```
Comment (usimd) [inline src/pi-gen-config.ts:101]: Shouldn't these two conditions be joined using `or`? 
```suggestion
    !/^[0-9]$/.test(config.compressionLevel) ||
```

**Review by usimd — COMMENTED** [review]
Thanks for your addition! Good idea to allow very aggressive compression, didn't even know of this setting 👍

Would you mind adding a test case to also hit the coverage?

File: src/pi-gen-config.ts (hunk 1 above)
Comment (dtcooper) [reply src/pi-gen-config.ts:101]: Pretty sure this is correct

Throw an error when

* (`compressionLevel` does NOT match `^[0-9]$`) and (`deployCompression` is `"xz"` and `compressionLevel` is not `"9e"`)

**Comment by dtcooper** [comment]
> Thanks for your addition! Good idea to allow very aggressive compression, didn't even know of this setting 👍
> 
> Would you mind adding a test case to also hit the coverage?

I'm by no means a typescript expert, but I will try!

File: src/pi-gen-config.ts (hunk 1 above)
Comment (dtcooper) [reply src/pi-gen-config.ts:101]: Woops. I think you're right. Let me take a second look.

**Comment by dtcooper** [comment]
Was having trouble writing a test...

**Comment by dtcooper** [comment]
Okay! My trouble with the tests were one of the current tests wasn't working properly. It _always_ threw an error because `stageList` wasn't properly converted to a list. I've fixed that test, tested for specific error cases, and added one for the new feature. Thanks!

**Comment by usimd** [comment]
Awesome, thanks so much for your effort!

I'll have a closer look tomorrow but I assume this should be fine and we can then merge quickly.

**Comment by dtcooper** [comment]
> Awesome, thanks so much for your effort!
> 
> I'll have a closer look tomorrow but I assume this should be fine and we can then merge quickly.

Great! Thanks!

**Review by usimd — APPROVED** [review]
Thanks again for your contribution!

**Pull request merged**